platform/ext/target/nordic_nrf/common/nrf9160/target_cfg.c

```diff
@@ -581,11 +586,19 @@ enum tfm_plat_err_t spu_periph_init_cfg(void)
      * - TWISx
      * - UARTEx
      */
+
+    /* UART0 is a secure peripheral, so we need to leave Serial-Box 0 as Secure.
+     * The UART Driver will configure it as non-secure when it uninitializes.
+     */
+#if !(defined(SECURE_UART1) && defined(CONFIG_TFM_SHARE_UART))
```

Comment:
Same comment as before, better to have a top level if with SECURE_UART to improve readability.